### DISCUSSION TITLE
Baseline calculation - CPU usage

### DISCUSSION THREAD

**Discussion by Lochnair** [sqm-autorate/sqm-autorate#87] [Operational Development]
Just a curiosity I noticed while watching the script CPU usage in htop (with threads shown).

The baseliner thread seems to take more CPU as the script runs for a few minutes. After the ratecontrol thread kicks in, it'll mostly hang around 2-3% on my box, with obvious higher usage when the selector kicks in.

But after running for a bit it'll gradually go up to around 21-24% and stay there, which is a bit higher than I'd like.

**Comment by Fail-Safe** [comment]
> But after running for a bit it'll gradually go up to around 21-24% and stay there, which is a bit higher than I'd like.

I am not seeing the same behavior on my box, but I do see where one of the threads uses considerably more CPU time than the rest.

![CPUtime](https://user-images.githubusercontent.com/10307870/148864258-61ef7af2-a3e1-4c9a-9ea5-a98d58b9fc47.png)

**Reply by Fail-Safe** [reply]
@Lochnair  Sorry if this is a dumb question, but are you somehow seeing the thread names themselves via htop?

**Reply by Lochnair** [reply]
Not a dumb question. It doesn't work out of the box. You gotta do this in the code to set the names: https://gist.github.com/Lochnair/781c5bf056ad61a85aecd0799f7f59a7

Htop doesn't show the thread names for me even with that option enabled sadly. So I'm using this one-liner to get them:
```bash
for f in $(find /proc/$(pgrep -f sqm-autorate)/task -name comm); do echo -n "$(echo $f | cut -d'/' -f5): "; cat "$f"; done
```

Example output:
```
7927: lua
7928: LanesTimer
7946: ping_receiver
7951: baseliner
7953: ping_sender
7958: peer_selector
8253: ratecontroller
```

**Reply by Lochnair** [reply]
I found that you can add the `COMM` column to the output in htop, and that'll show the names set as well.

**Comment by dlakelan** [comment]
It's strange because the baseline calculator doesn't do that much. The ratecontroller calls sort for example, but there's not much calculation in the baseliner. Do you have logging set to a high level? Perhaps it's just printing a lot of stuff?

**Reply by Lochnair** [reply]
Exactly, that's why it baffles me. It shouldn't use much at all, and yet it does.

And no I have my log level set to INFO. Besides the baseline calculator only ever logs anything itself if `enable_verbose_output` is set to `true` in the code.
